plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorExtFactory.java

```diff
@@ -77,39 +64,60 @@ public Mediator createSpecificMediator(OMElement elem, Properties properties) {
-					} else {
-						handleException("Specified class for the hashGenerator is not a "
-								+ "DigestGenerator. It *must* implement "
-								+ "org.wso2.carbon.mediator.cache.digest.DigestGenerator interface");
-					}
-				} catch (ClassNotFoundException e) {
-					handleException("Unable to load the hash generator class", e);
-				} catch (IllegalAccessException e) {
-					handleException("Unable to access the hash generator class", e);
-				} catch (InstantiationException e) {
-					handleException("Unable to instantiate the hash generator class", e);
-				}
-			}
-
             OMAttribute timeoutAttr = elem.getAttribute(ATT_TIMEOUT);
             if (timeoutAttr != null && timeoutAttr.getAttributeValue() != null) {
                 cacheMediator.setTimeout(Long.parseLong(timeoutAttr.getAttributeValue()));
             } else {
                 cacheMediator.setTimeout(DEFAULT_TIMEOUT);
             }
-
+            
             OMAttribute maxMessageSizeAttr = elem.getAttribute(ATT_MAX_MSG_SIZE);
             if (maxMessageSizeAttr != null && maxMessageSizeAttr.getAttributeValue() != null) {
-                cacheMediator.setMaxMessageSize(Integer.parseInt(maxMessageSizeAttr.getAttributeValue()));
+            	cacheMediator.setMaxMessageSize(Integer.parseInt(maxMessageSizeAttr.getAttributeValue()));
+            }
+
+            for (Iterator itr = elem.getChildrenWithName(PROTOCOL_Q); itr.hasNext();) {
+                OMElement implElem = (OMElement) itr.next();
+                OMAttribute typeAttr = implElem.getAttribute(ATT_TYPE);
+                if (typeAttr != null && typeAttr.getAttributeValue() != null) {
+                    String protocolType = typeAttr.getAttributeValue();
+                    cacheMediator.setProtocolType(protocolType);
+                }
+                for(Iterator itrChildren = implElem.getChildrenWithLocalName("methods"); itrChildren.hasNext();) {
+                	OMElement child = (OMElement)itrChildren.next();
+                	cacheMediator.setHTTPMethodsToCache(child.getText().split(","));
+                }
+                for(Iterator itrChildren = implElem.getChildrenWithLocalName("headersToExcludeInHash"); itrChildren.hasNext();) {
+                	OMElement child = (OMElement)itrChildren.next();
+                	cacheMediator.setHeadersToExcludeInHash(child.getText().split(","));
+                }
+                for(Iterator itrChildren = implElem.getChildrenWithLocalName("responseCodes"); itrChildren.hasNext();) {
+                	OMElement child = (OMElement)itrChildren.next();
+                	cacheMediator.setResponseCodes(child.getText());
+                }
+                for(Iterator itrChildren = implElem.getChildrenWithLocalName("hashGenerator"); itrChildren.hasNext();) {
+                	OMElement child = (OMElement)itrChildren.next();
+
+        			if (child.getText() != null && child.getText() != null) {
```

Comment:
What is the logic behind this: (child.getText() != null && child.getText() != null)